PublicAPI/Microsoft.ApplicationInsights.dll/net45/PublicAPI.Unshipped.txt

```diff
@@ -33,5 +33,237 @@ Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.IHeartbeatPro
 Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.IHeartbeatPropertyManager.IsHeartbeatEnabled.get -> bool
 Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.IHeartbeatPropertyManager.IsHeartbeatEnabled.set -> void
 Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.IHeartbeatPropertyManager.SetHeartbeatProperty(string propertyName, string propertyValue = null, bool? isHealthy = null) -> bool
-Microsoft.ApplicationInsights.DataContracts.PageViewTelemetry.Id.get -> string
-Microsoft.ApplicationInsights.DataContracts.PageViewTelemetry.Id.set -> void
\ No newline at end of file
+Microsoft.ApplicationInsights.Metric
```

Comment:
historically we keep namespace `Microsoft.ApplicationInsights` clean. So developer has only option in intelliSense after typing `Microsoft.ApplicaitonInsights` to create a `TelemetryClient`. I remember we looked at implementation and only thing in root namespace was change in `TelemetryClient`. Is my recollection incorrect?

Reply:
Of course I may not have the right historical perspective.
However, this is not quite correct:
 >  I remember we looked at implementation and only thing in root namespace was change in TelemetryClient.

The public API service is identical to what we reviewed at the end of 2017 (barring the types that we made internal this week as a follow up to @Dmitry-Matveev 's feedback). At the time, I made a set of examples available, which are now also in the Core repo: 

https://github.com/Microsoft/ApplicationInsights-dotnet/blob/feature/PreAggMetrics.Master/Test/Microsoft.ApplicationInsights.Test/Shared/Metrics/MetricsExamples.cs

In that set of examples, you can find use case that get progressively more advanced. The goal was to enable the use cases from example 1 to be discovered via Intellisense without the need for additional namespace imports. In order to achieve that, the base namespace includes all types that may appear as parameters to the metric related TelemetryClient methods.

Please let me know if this answers your question.
If you require adjustments to these APIs, please request precisely the changes you would like to see so that I can make them.
Thank you.